packages/hooks/src/useGroupedFiatTransactions.ts

```diff
@@ -0,0 +1,121 @@
+            longcode: 'Win payout if AUD/JPY is strictly lower than entry spot at 1 minute after contract start time.',
+            payout: 1,
+            purchase_time: 1470637295,
+            reference_id: 13692993001,
+            shortcode: 'PUT_FRXAUDJPY_1_1470637295_1470637355_S0P_0',
+            transaction_id: 13693003421,
+            transaction_time: 1685109944,
+        },
+        {
+            action_type: 'deposit',
+            amount: 2,
+            app_id: {},
+            balance_after: 44.24,
+            contract_id: {},
+            longcode: 'Payment from Binary Services Ltd Apr 2017',
+            payout: {},
+            reference_id: {},
+            shortcode: {},
+            transaction_id: 17494117541,
+            transaction_time: 1685009944,
+        },
+        {
+            action_type: 'transfer',
+            amount: 5.55,
+            from: {
+                loginid: 'CR90000000',
+                type: 'trading',
+            },
+            to: {
+                loginid: 'CR90000003',
+                type: 'wallet',
+            },
+            app_id: {},
+            balance_after: 5.55,
+            contract_id: {},
+            longcode: 'Transfer from <> to <>. Includes fees',
+            payout: {},
+            reference_id: {},
+            shortcode: {},
+            transaction_id: 17494415489,
+            transaction_time: 1685009944,
+        },
+        {
+            action_type: 'withdrawal',
+            amount: -5.55,
+            app_id: {},
+            balance_after: 0,
+            contract_id: {},
+            longcode: 'Account closed. Please contact customer support for assistance.',
+            payout: {},
+            reference_id: {},
+            shortcode: {},
+            transaction_id: 17494415481,
+            transaction_time: 1684009944,
+        },
+        {
+            action_type: 'transfer',
+            amount: 5.55,
+            from: {
+                loginid: 'CR90000000',
+                type: 'trading',
+            },
+            to: {
+                loginid: 'CRW1000',
+                type: 'wallet',
+            },
+            app_id: {},
+            balance_after: 5.55,
+            contract_id: {},
+            longcode: 'Transfer from <> to <>. Includes fees',
+            payout: {},
+            reference_id: {},
+            shortcode: {},
+            transaction_id: 17494415482,
+            transaction_time: 1684009944,
+        },
+    ];
+
+    const grouped_transactions = _.pickBy(
+        // TODO replace mock_transactions with transactions in the line below
+        _.groupBy(mock_transactions, transaction => {
+            return transaction.transaction_time
+                ? moment(transaction.transaction_time * 1000)
+                      .startOf('day')
+                      .format('DD MMM YYYY')
+                : null;
+        }),
+        (value, key) => key !== null
+    );
```

Comment:
@rostislav-deriv A bit confused with this hook 👀 I don't think we really need this 😬

`statement` is a paginated API, So for each call we get the next page, Then how is the grouping gonna work? Are you going to group each fetch separately or the whole list? I believe grouping and fetching are separated for this use case.
Grouping should be the UI's responsibility, It should be a generic option on the list component.

Reply:
@farzin-deriv, good point. As I remember, the transaction list is to be a lazy-loaded scroll view, with transactions coming from the API in portions, so the fetching is on-demand and is to be done in the component using `useFetch` if I'm not mistaken. So is the grouping, which is to be performed every time the transaction list is changed. The question is though - where to store the transactions 🤔 The most obvious way is to store them in component as state and change this hook to accept the list and return the grouped object.
P.S. I don't think the grouping should be done inside the component because it requires the dependencies, so it's either a hook with parameters (for now), or a util in `@deriv/utils` (later).

Reply:
> where to store the transactions

@rostislav-deriv You can use the `keepPreviousData` option on `useFetch` 👀 
https://tanstack.com/query/v4/docs/react/guides/paginated-queries

> I don't think the grouping should be done inside the component

We have `DataList` generic component and grouping should be the responsibility of that component.

You can check the [ListBuilder](https://github.com/regentmarkets/flutter-otc-cashier/blob/dev/lib/features/core/presentation/widgets/list_builder.dart) widget and it's `groupBy` property we have in the mobile codebase to get the idea

<img width="535" alt="image" src="https://github.com/binary-com/deriv-app/assets/72082844/41546093-1b8a-42a9-8e6d-651241bec2ea">

Reply:
For now, I have changed the code to fetching via `useFetch` inside the component and using the hook for grouping but the hook has to go in favour of the library for doing the generic grouping so that we don't fill up the hooks package with some niche dependencies for random stuff